### PR TITLE
chore(codecs): Implement `StandardEncodingsMigrator`

### PR DESCRIPTION
**Pull request by pablosichert**
Context: https://github.com/vectordotdev/vector/pull/12136#discussion_r846248439 / https://github.com/vectordotdev/vector/pull/12133#discussion_r846354046.

### CODE REVIEW

File: src/sinks/util/encoding/mod.rs [hunk 1]
```diff
@@ -104,12 +110,24 @@ pub trait Encoder<T> {
...
 
-impl<E, T> Encoder<T> for Arc<E>
-where
-    E: Encoder<T>,
-{
-    fn encode_input(&self, input: T, writer: &mut dyn io::Write) -> io::Result<usize> {
-        (**self).encode_input(input, writer)
+#[cfg(feature = "codecs")]
+impl Encoder<Vec<Event>> for (Transformer, crate::codecs::Encoder<()>) {
+    fn encode_input(&self, events: Vec<Event>, writer: &mut dyn io::Write) -> io::Result<usize> {
```
Comment (lukesteensen) [inline src/sinks/util/encoding/mod.rs:115]: It's not super obvious to me how this is related to the primary change.

**Review by lukesteensen — APPROVED** [review]
Looks reasonable to me

**Review by jszwedko — APPROVED** [review]
Similar question to Luke, otherwise 👍

File: src/sinks/util/encoding/mod.rs (hunk 1 above)
Comment (pablosichert) [reply src/sinks/util/encoding/mod.rs]: Agree that this wasn't super obvious 😅 I factored that part out into a separate PR: https://github.com/vectordotdev/vector/pull/12178.

This has just been a piece that is used by many sinks that use `StandardEncodings`.

**Comment by pablosichert** [comment]
Calling for another round of reviews since I added `StandardEncodingsWithFramingMigrator`, next to the existing `StandardEncodingsMigrator`.

**Review by lukesteensen — APPROVED** [review]

**Pull request merged**